tests/components/smarttub/test_climate.py

```diff
@@ -0,0 +1,102 @@
+"""Test the SmartTub climate platform."""
+from unittest.mock import Mock
+
+from homeassistant.components.climate.const import (
+    ATTR_CURRENT_TEMPERATURE,
+    ATTR_HVAC_ACTION,
+    ATTR_HVAC_MODE,
+    ATTR_HVAC_MODES,
+    ATTR_MAX_TEMP,
+    ATTR_MIN_TEMP,
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_IDLE,
+    DOMAIN as CLIMATE_DOMAIN,
+    HVAC_MODE_HEAT,
+    SERVICE_SET_HVAC_MODE,
+    SERVICE_SET_TEMPERATURE,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.components.smarttub.climate import async_setup_entry
+from homeassistant.components.smarttub.const import (
+    DEFAULT_MAX_TEMP,
+    DEFAULT_MIN_TEMP,
+    DOMAIN,
+    SMARTTUB_CONTROLLER,
+    UNSUB_UPDATE_LISTENER,
+)
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    ATTR_SUPPORTED_FEATURES,
+    ATTR_TEMPERATURE,
+)
+
+from tests.common import MockConfigEntry
+
+
+async def test_async_setup_entry(hass, controller):
+    """Test async_setup_entry."""
+
+    entry = MockConfigEntry(unique_id="ceid1")
+    async_add_entities = Mock()
+    hass.data[DOMAIN] = {
+        entry.unique_id: {
+            SMARTTUB_CONTROLLER: controller,
+            UNSUB_UPDATE_LISTENER: lambda: None,
+        },
+    }
+
+    ret = await async_setup_entry(hass, entry, async_add_entities)
```

Comment:
Use the standard interfaces in `hass.config_entries` to setup the entry and then check the state machine to verify the entities exist

Reply:
That's what I do in `test_thermostat`. Does that mean this test is redundant and I should remove it?

Reply:
Yes, coverage is still 100% without this test. Removing.